src/ReverseProxy/Delegation/HttpSysDelegatorMiddleware.cs

```diff
@@ -24,15 +24,20 @@ public HttpSysDelegatorMiddleware(
         HttpSysDelegator delegator,
         IRandomFactory randomFactory)
     {
-        _next = next ?? throw new ArgumentNullException(nameof(next));
-        _logger = logger ?? throw new ArgumentNullException(nameof(logger));
-        _delegator = delegator ?? throw new ArgumentNullException(nameof(delegator));
-        _randomFactory = randomFactory ?? throw new ArgumentNullException(nameof(randomFactory));
+        ArgumentNullException.ThrowIfNull(next);
+        _next = next;
+        ArgumentNullException.ThrowIfNull(logger);
+        _logger = logger;
+        ArgumentNullException.ThrowIfNull(delegator);
+        _delegator = delegator;
+        ArgumentNullException.ThrowIfNull(randomFactory);
```

Comment:
Let's try to move all the exception check before the assignment 

```suggestion
        ArgumentNullException.ThrowIfNull(logger);
        ArgumentNullException.ThrowIfNull(delegator);
        ArgumentNullException.ThrowIfNull(randomFactory);
        _next = next;
        _logger = logger;
        _delegator = delegator;
```